src/dicomweb_client/uri.py

```diff
@@ -0,0 +1,358 @@
+
+    @property
+    def series_instance_uid(self) -> Optional[str]:
+        """Returns the Series UID, if available."""
+        return self._series_instance_uid
+
+    @property
+    def sop_instance_uid(self) -> Optional[str]:
+        """Returns the Instance UID, if available."""
+        return self._instance_uid
+
+    @property
+    def type(self) -> URIType:
+        """The `URIType` of DICOM resource referenced by the object."""
+        if self.study_instance_uid is None:
+            return URIType.SERVICE
+        elif self.series_instance_uid is None:
+            return URIType.STUDY
+        elif self.sop_instance_uid is None:
+            return URIType.SERIES
+        return URIType.INSTANCE
+
+    def study_uri(self) -> 'URI':
+        """Returns `URI` for the DICOM Study within this object."""
+        if self.type == URIType.SERVICE:
+            raise ValueError('Cannot get a Study URI from a Base (DICOMweb '
+                             'service) URL.')
+        return URI(self.base_url, self.study_instance_uid)
+
+    def series_uri(self) -> 'URI':
+        """Returns `URI` for the DICOM Series within this object."""
+        if self.type in (URIType.SERVICE, URIType.STUDY):
+            raise ValueError(
+                f'Cannot get a Series URI from a {self.type!r} URI.')
+        return URI(self.base_url, self.study_instance_uid,
+                   self.series_instance_uid)
+
+    def update(self,
+               base_url: Optional[str] = None,
+               study_instance_uid: Optional[str] = None,
+               series_instance_uid: Optional[str] = None,
+               sop_instance_uid: Optional[str] = None) -> 'URI':
+        """Creates a new `URI` object based on the current one.
+
+        Replaces the specified `URI` components in the current `URI` to create
+        the new one.
+
+        Parameters
+        ----------
+        base_url: str, optional
+            DICOMweb service HTTP[S] URL to use in the new `URI` or `None` if the
+            `base_url` from the current `URI` should be used.
+        study_instance_uid: str, optional
+            Study Instance UID to use in the new `URI` or `None` if the
+            `study_instance_uid` from the current `URI` should be used.
+        series_instance_uid: str, optional
+            Series Instance UID to use in the new `URI` or `None` if the
+            `series_instance_uid` from the current `URI` should be used.
+        sop_instance_uid: str, optional
+            SOP Instance UID to use in the new `URI` or `None` if the
+            `sop_instance_uid` from the current `URI` should be used.
+
+        Returns
+        -------
+        URI
+          The newly constructed `URI` object.
+
+        Raises
+        ------
+        ValueError
+            If the new `URI` is invalid (e.g., if only the SOP Instance UID is
+            specified, but the Series Instance UID is missing in the current
+            `URI`).
+        """
+        return URI(
```

Comment:
In case we want to support suffixes, we would need to add them to the newly constructed URI as well.

Reply:
Done.